discovery-provider/src/utils/ipfs_lib.py

```diff
@@ -168,41 +168,57 @@ def cat(self, multihash):
             raise  # error is of type ipfshttpclient.exceptions.TimeoutError
 
     def multihash_is_directory(self, multihash):
+        # First attempt to cat multihash locally.
         try:
-            # attempt to cat single byte from CID to determine if dir or file
+            # If cat successful, multihash is not directory.
             self._api.cat(multihash, 0, 1)
             return False
         except Exception as e:  # pylint: disable=W0703
             if "this dag node is a directory" in str(e):
-                logger.warning(f'IPFSCLIENT | Found directory {multihash}')
+                logger.warning(f"IPFSCLIENT | Found directory {multihash}")
                 return True
 
-        # Attempt to retrieve from cnode gateway endpoints
+        # Attempt to retrieve from cnode gateway endpoints.
```

Comment:
@SidSethi there's no way this endpoint will cause the cnode to error and crash right?

Reply:
pretty much everything in that route is try-catched so don't think so. do you have any potential failure cases in mind?